File: 743/743.py
class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        result = []
        for i in range(1, n + 1):
            pushMe = ''
            divBy3 = i % 3 == 0
            divBy5 = i % 5 == 0
            
            if divBy3:
                pushMe += 'Fizz'
            if divBy5:
                pushMe += 'Buzz'
            if not divBy3 and not divBy5:
                pushMe += str(i)
            result.append(pushMe)
            
        return result

File: 743/743.rb
=begin
Given an integer n, return a string array answer (1-indexed) where:

answer[i] == "FizzBuzz" if i is divisible by 3 and 5.
answer[i] == "Fizz" if i is divisible by 3.
answer[i] == "Buzz" if i is divisible by 5.
answer[i] == i if non of the above conditions are true.
=end

# @param {Integer} n
# @return {String[]}
def fizz_buzz(n)
    output = []
    
    (1..n).each do |number|
        result = ''
        div_by_3 = number % 3 == 0
        div_by_5 = number % 5 == 0
        
        result += 'Fizz' if div_by_3
        result += 'Buzz' if div_by_5
        result += number.to_s unless div_by_3 || div_by_5
        
        output << result
    end
    output
end
